src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -82,71 +85,36 @@ public static Checksum Create(ReadOnlySpan<Checksum> hashes)
     }
 
     public static Checksum Create(ArrayBuilder<Checksum> checksums)
-    {
-        using var stream = SerializableBytes.CreateWritableStream();
-
-        using (var writer = new ObjectWriter(stream, leaveOpen: true))
+        => Create(checksums, static (checksums, writer) =>
         {
             foreach (var checksum in checksums)
                 checksum.WriteTo(writer);
-        }
```

Comment:
all these helpers now defer to the helper that takes a callback and passed it an ObjectWriter to write data into (which itself is then checksumed).  This is teh same logic as before, just with teh common code of creating the objectwriter and checksum'ing it pushed to the helper.